front/components/items/FindOrCancelAppointment/index.tsx

```diff
@@ -0,0 +1,198 @@
+import { useFormik } from 'formik';
+import React, { useEffect, useRef, useState } from 'react';
+import * as Yup from 'yup';
+import Paragraph from '../Paragraph';
+import Button from '../Button';
+import { strapiAppointmentGet } from 'lib/strapi/appointments/get';
+import { filtersAppointmentCode } from './models';
+import { strapiAppointmentQuery } from 'lib/strapi/appointments/queryType';
+import { strapiAppointmentPut } from 'lib/strapi/appointments/put';
+import WrapperColumn from 'components/wrappers/WrapperColumn';
+
+const FindOrCancelAppointment = () => {
+    const [data, setData] = useState<string>();
+    const [isSubmited, setIsSubmited] = useState(false);
+    const [isSearching, setIsSearching] = useState(false);
+    const [isCanceling, setIsCanceling] = useState(false);
+    const [messageOfCancelingProperlyDone, setMessageOfCancelingProperlyDone] = useState<string>();
+    const properCodeInput = useRef<HTMLInputElement>(null);
+    const [foundAppointment, setFoundAppoinment] = useState<strapiAppointmentQuery[]>();
+
+    const filters: filtersAppointmentCode = {
+        appointment_code: {
+            $eq: data,
+        },
+    };
+
+    const validateCode = () => {
+        if (properCodeInput.current)
+            properCodeInput.current.value = properCodeInput.current.value.replace(/[^0-9]/g, '');
+    };
+
+    const canceledAppointment = {
+        data: {
+            appointment_code: null,
+            is_booked: false,
+        },
+    };
+
+    const canceledAppointmentId = () => {
```

Comment:
I'd rename function to maybe 'getCanceledAppointmentId or sth similar. RN it sounds like variable, not function